modules/300-prometheus/openapi/config-values.yaml

```diff
@@ -112,6 +112,19 @@ properties:
       The class of the Ingress controller used for Grafana/Prometheus.
 
       An optional parameter. By default, the `modules.ingressClass` global value is used.
+  externalLabels:
+    type: object
+    x-examples:
+      - prometheus: dev
+        prometheus_replica: ${HOSTNAME}
+        env: $MY_ENV
+    description: |
+      The set of external labels to add to the metrics.
+      
+      It is possible to expand [environment variables](https://prometheus.io/docs/prometheus/latest/feature_flags/#expand-environment-variables-in-external-labels) in external labels.
+    additionalProperties:
+      type: string
+      maxLength: 63
```

Comment:
Why is this limited? AFAIK labels in Prometheus can be any length. This is also not protecting users from overcome the limit by expanding the env variable, e.g., `abcabcabcabcabcabcabcabcabcabcabcabcabcabcabcabcabcabc $HOSTNAME`

Reply:
Just some sort of precaution, I guess :)) I wanted to impose some regex but wasn't able to come up with something really useful because of env var expanding. Just wonder what happens if we use something really nasty, what Prometheus limits can we reach..

Reply:
Prometheus does not have a limit. Any string is valid. The only problem is that metrics storage on the other end may struggle with huge labels, but it should not be a real problem.

Reply:
length constrained has been lifted